### PR TITLE
Category page

### PR DESCRIPTION
**Pull request by Pomax**
Closes https://github.com/mozilla/foundation.mozilla.org/issues/1821

Adds a product category as snippet-administratable thing, with default categories based on the comp, and which are used to generate the nav menu now, linking to category pages. Each product can have multiple categories - when on a category page the relevate cat. will be menu-highlighted, and when clicking through to a product no cat is highlighted.

This PR also updates the fake data generator so that it generates 20 products, each assigned to at least one, but possibly more categories.

### CODE REVIEW

File: network-api/networkapi/buyersguide/views.py [hunk 1]
```diff
@@ -20,21 +20,39 @@
...
+
+@login_required
+def category_view(request, categoryname):
+    category = BuyersGuideProductCategory.objects.get(name__iexact=categoryname)
+    products = [p.to_dict() for p in Product.objects.filter(product_category__in=[category]).distinct()]
+    return render(request, 'category_page.html', {
+        'categories': BuyersGuideProductCategory.objects.all(),
+        'category': category,
+        'products': products,
+        'mediaUrl': settings.MEDIA_URL,
```
Comment (Pomax) [inline network-api/networkapi/buyersguide/views.py]: this is not enough to get images to work properly.

File: network-api/networkapi/wagtailpages/models.py [hunk 2]
```diff
@@ -911,7 +911,6 @@ def get_context(self, request):
         # We need to expose MEDIA_URL so that the s3 images will show up properly
         # due to our custom image upload approach pre-wagtail
         context = super(Homepage, self).get_context(request)
-        print(settings.MEDIA_URL)
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/models.py]: spotted this while figuring out why MEDIA_URL doesn't include the AWS directory, so... remove'd

**Comment by kristinashu** [comment]
@Pomax what url can I see this at? I'll do a design review too.

**Comment by kristinashu** [comment]
This is good to push but I will open a follow up ticket to update the font styles:
- Category title .h3-heading
- Disclaimer .body with italic 
- Disclaimer should be moved to the bottom of the page
- Product company .h6-heading-uppercase
- Product name .h5-heading

category 2x

File: source/sass/buyers-guide/views/category.scss [hunk 3]
```diff
@@ -0,0 +1,89 @@
+.category-listing {
```
Comment (gvn) [inline source/sass/buyers-guide/views/category.scss]: Most of this page layout should be achievable with Bootstrap and other existing type classes (see https://foundation.mozilla.org/en/style-guide/), but I'll leave it up to you if you want to refactor given time constraints.

File: network-api/networkapi/buyersguide/urls.py [hunk 4]
```diff
@@ -5,6 +5,7 @@
 urlpatterns = [
     url(r'^$', views.buyersguide_home, name='buyersguide-home'),
     url(r'^about/', views.about_view, name='about-view'),
+    url(r'^category/(?P<categoryname>[\w\W]+)/', views.category_view, name='category-view'),
```
Comment (gvn) [inline network-api/networkapi/buyersguide/urls.py]: Might want to use `categories/` for consistency with the recently renamed `products/` route fragment. @cadecairos may have thoughts.

Comment (cadecairos) [reply network-api/networkapi/buyersguide/urls.py]: yes, lets keep the resource names in our routes pluralized

**Comment by kristinashu** [comment]
Any chance we can push this live today (even though Pomax is off today) so we don't have to wait until Monday? Planning to do a bunch of content entry tomorrow.

File: source/sass/buyers-guide/views/category.scss (hunk 3 above)
Comment (Pomax) [reply source/sass/buyers-guide/views/category.scss]: it's  good idea, although I will need to learn myself some bootstrap before I can do this, so I'll file it as a followup: https://github.com/mozilla/foundation.mozilla.org/issues/1924

File: network-api/networkapi/buyersguide/urls.py (hunk 4 above)
Comment (Pomax) [reply network-api/networkapi/buyersguide/urls.py]: :+1:

**Pull request merged**